in-person-payments-example/src/main/java/com/adyen/ipp/api/InPersonPaymentsController.java

```diff
@@ -0,0 +1,207 @@
+package com.adyen.ipp.api;
+
+import com.adyen.model.nexo.*;
+import com.adyen.ipp.ApplicationProperty;
+import com.adyen.ipp.model.*;
+import com.adyen.ipp.request.*;
+import com.adyen.ipp.response.*;
+import com.adyen.ipp.service.*;
+import com.adyen.ipp.util.IdUtility;
+import com.adyen.service.exception.ApiException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+
+/**
+ * REST controller for using Adyen checkout API
+ */
+@RestController
+@RequestMapping("/api")
+public class InPersonPaymentsController {
+    private final Logger log = LoggerFactory.getLogger(InPersonPaymentsController.class);
+
+    private final ApplicationProperty applicationProperty;
+
+    @Autowired
+    private PosAbortService posAbortService;
+
+    @Autowired
+    private PosPaymentService posPaymentService;
+
+    @Autowired
+    private PosReversalService posReversalService;
+
+    @Autowired
+    private PosTransactionStatusService posTransactionStatusService;
+
+    @Autowired
+    private TableService tableService;
+
+    private String saleId;
+
+    private String poiId;
+
+    @Autowired
+    public InPersonPaymentsController(ApplicationProperty applicationProperty) {
+
+        this.applicationProperty = applicationProperty;
+        saleId = applicationProperty.getSaleId();
+        poiId = applicationProperty.getPoiId();
+    }
+
+    @PostMapping("/create-payment")
+    public ResponseEntity<CreatePaymentResponse> createPayment(@RequestBody CreatePaymentRequest request) throws IOException, ApiException {
+        Table table = tableService.getTables().stream()
+                .filter(t -> t.getTableName().equals(request.getTableName()))
+                .findFirst()
+                .orElse(null);
+
+        if (table == null) {
+            return ResponseEntity
+                    .status(404)
+                    .body(new CreatePaymentResponse()
+                            .result("failure")
+                            .refusalReason("Table " + request.getTableName() + " not found"));
+        }
+
+        String serviceId = IdUtility.getRandomAlphanumericId(10);
+
+        table.getPaymentStatusDetails().setServiceId(serviceId);
+        table.setPaymentStatus(PaymentStatus.PaymentInProgress);
+
+        try {
```

Comment:
Minor comment as the approach is fine, but I would start the `try-catch` as the beginning of the method (line 58) to make sure that any error (even when code is added later) results in the same exception handling
```
            log.error(e.toString());
            table.setPaymentStatus(PaymentStatus.NotPaid);
            throw e;
```